graphql/course/fields.ts

```diff
@@ -71,4 +71,39 @@ export class ExtendedFieldsCourseResolver {
         const student = await getSessionStudent(context, studentId);
         return (await prisma.course_instructors_student.count({ where: { courseId: course.id, studentId: student.id } })) > 0;
     }
+
+    @Query((returns) => [Course])
+    @Authorized(Role.ADMIN, Role.OWNER, Role.INSTRUCTOR)
+    async templateCourses(
+        @Arg('studentId', { nullable: true }) studentId: number,
+        @Arg('search') search: string,
+        @Arg('take', () => GraphQLInt) take,
+        @Arg('skip', () => GraphQLInt, { nullable: true }) skip: number = 0
+    ) {
+        const courseSearchFilters = await courseSearch(search);
+        const courses = await prisma.course.findMany({
+            where: {
+                OR: [
+                    {
+                        AND: [
+                            {
+                                course_instructors_student: {
+                                    some: {
+                                        studentId: studentId,
+                                    },
+                                },
+                            },
+                            courseSearchFilters,
```

Comment:
the search will not search shared courses

Reply:
I guess you want AND [ OR [ own, shared ], search ]

Reply:
As @realmayus explained to me, a course shall be returned as a template if:
- shared: true or
- the student is instructor of the course.

This or is handled by the outer OR in line 86. The inner AND in line 88 handles the second case along with the given search parameters. So it looks correct to me, but of course I might be wrong.

Reply:
Yeah well the search will currently only search the own courses, it should also search the shared courses.

Reply:
> Yeah well the search will currently only search the own courses, it should also search the shared courses.

The query will also find shared courses since the shared:true condition is part of the outer OR block.

@realmayus What do you think?

Reply:
@paul-jaworski Yes, but it will return ***all*** shared courses, but we want to be able to filter (i.e. search within) the shared courses as well. If there are 1000 shared courses, it would return all 1000 of them, even if we have entered a very restrictive search query.

Currently the query does the following:
```
if course.student == student {
    if searchFilters.match(course) == true {
        return course
    }
} else if course.isShared {
    return course
}
```
what we want is:
```
if searchFilters.match(course) == true {
    if course.student == student {
        return course
    } else if course.isShared {
        return course
    }
}
```